### PR TITLE
Cherry pick DYN- 3814 and DYN-3822 to 2.12.0

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

* Cherry pick DYN- 3814 and DYN-3822 to 2.12.0

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/DynamoCoreWpfTests/PreferencesViewModelTests.cs [hunk 1]
```diff
@@ -20,10 +24,59 @@ public void SelectedPackagePathForInstall_Setter()
...
+            // Check that the selection has updated
+            Assert.AreEqual(@"D:\", preferencesVM.SelectedPackagePathForInstall);
+
+            // Remove path
+            preferencesVM.PackagePathsViewModel.RootLocations.Remove(@"D:\");
+
+            // Check that the selection has changed and is not empty
+            var selection = preferencesVM.SelectedPackagePathForInstall;
+            Assert.AreNotEqual(@"D:\", selection);
+            Assert.IsFalse(string.IsNullOrEmpty(selection));
```
Comment (aparajit-pratap) [inline test/DynamoCoreWpfTests/PreferencesViewModelTests.cs:71]: What is `selection` in this case - is it appdata?

**Comment by mjkkirschner** [comment]
testing this now in master.

File: test/DynamoCoreWpfTests/PreferencesViewModelTests.cs (hunk 1 above)
Comment (sm6srw) [reply test/DynamoCoreWpfTests/PreferencesViewModelTests.cs:71]: yes

Comment (aparajit-pratap) [reply test/DynamoCoreWpfTests/PreferencesViewModelTests.cs:71]: Can we add an assertion verifying that that is the case?

Comment (aparajit-pratap) [reply test/DynamoCoreWpfTests/PreferencesViewModelTests.cs:71]: Actually, since this is a cherry-pick, never mind. Minor point.

Comment (sm6srw) [reply test/DynamoCoreWpfTests/PreferencesViewModelTests.cs:71]: Sure

**Pull request merged**